assets/js/modules/analytics/common/create-account-field.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * CreateAccountField component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	Input,
+	TextField,
+} from '../../../material-components';
+import classnames from 'classnames';
+
+const CreateAccountField = ( {
+	validationIssues,
+	setValidationIssues,
+	objectValue,
+	objectSetter,
+	name,
+	label,
+} ) => {
+	return (
+		<TextField
+			className={ classnames(
+				'mdc-text-field',
+				{ 'mdc-text-field--error': validationIssues[ name ] }
+			) }
+			label={ label }
+			name={ name }
+			onChange={ ( e ) => {
+				setValidationIssues( {
+					...validationIssues,
+					[ name ]: '' === e.target.value,
+				} );
```

Comment:
Setting the validation issues from here results in duplicated code and can be simplified. We can rely on the `useEffect` hook in the main component to re-check validation whenever one of these four values changes. No need to re-handle it here. It will also nicely reduce the amount of props necessary.

Reply:
Great suggestion! Switched to `useEffect` in https://github.com/google/site-kit-wp/pull/1389/commits/f8296c5a2fd1f65e1721662100349bef99872c96.